force-app/main/default/classes/BDI_ContactService.cls

```diff
@@ -343,7 +393,25 @@ public with sharing class BDI_ContactService {
                 }
 
                 di.put(strCx + 'Imported__c', con.id);
-                di.put(strCx + 'ImportStatus__c', con.id != null ? bdi.statusMatchedExisting() : bdi.statusMatchedNew());
+
+                // Determine correct import status based on the type of matching and whether it matched on an existing
+                // record
+                String importStatus;
```

Comment:
Do you think we can make this change a little cleaner?  we could extract this change into its own method and also extract these guard clauses that we use to determine the kind of contact into smaller methods so we improve readability.

Reply:
Made this update